cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperSetOperator.java

```diff
@@ -83,7 +83,7 @@ public static boolean needsRollingUpdate(StatefulSetDiff diff) {
         String cluster = sts.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
         for (int i = 0; i < replicas; i++) {
             Pod pod = podOperations.get(sts.getMetadata().getNamespace(), KafkaResources.zookeeperPodName(cluster, i));
-            zkRoll |= podRestart.test(pod);
+            zkRoll |= podRestart.apply(pod) != null && !podRestart.apply(pod).isEmpty();
```

Comment:
Don't assume that calling `podRestart` is cheap enough that you can call it twice; call it once and assign the result to a local var . And why not log the returned reasons?